### PR TITLE
Annotate GitExtUtils project

### PR DESCRIPTION
**Pull request by drewnoakes**
Follows #8771.

This moves `Annotations.cs` and `Strings.cs` into the `GitExtUtils` project and links to them from `GitCommands`.

### CODE REVIEW

File: GitExtUtils/GitUI/ThreadHelper.cs [hunk 1]
```diff
@@ -17,9 +17,9 @@ public static class ThreadHelper
...
 
-        private static JoinableTaskContext _joinableTaskContext;
-        private static JoinableTaskCollection _joinableTaskCollection;
-        private static JoinableTaskFactory _joinableTaskFactory;
+        private static JoinableTaskContext _joinableTaskContext = null!;
+        private static JoinableTaskCollection _joinableTaskCollection = null!;
+        private static JoinableTaskFactory _joinableTaskFactory = null!;
 
         public static JoinableTaskContext JoinableTaskContext
         {
```
[the remark this answers is not in the document]
Comment (drewnoakes) [reply GitExtUtils/GitUI/ThreadHelper.cs:25]: It's not correct, but it is pragmatic. Those JTC/JTF properties are used _everywhere_ and the assumption is they are basically never null. Assigning `null!` is a way of faking a non-null assignment without actually doing so. It's possible that having a run-time null check in the property makes sense and is a decent middle ground. I'll try that.

File: GitExtUtils/GitUI/Theming/LightnessCorrection.cs [hunk 2]
```diff
@@ -14,7 +14,13 @@ public LightnessCorrection(Bitmap bmp)
 
         protected override void ExecuteRaw()
         {
+            if (BgraValues == null)
```
[the remark this answers is not in the document]
Comment (drewnoakes) [reply GitExtUtils/GitUI/Theming/LightnessCorrection.cs:17]: This is going to take some getting used to...

Wonder if there's an analyzer we can use to enforce this.

[the remark this answers is not in the document]
Comment (RussKie) [reply GitExtUtils/GitUI/Theming/LightnessCorrection.cs:17]: @sharwell would be the man to ask :)

File: GitExtUtils/GitUI/ThreadHelper.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (RussKie) [reply GitExtUtils/GitUI/ThreadHelper.cs:25]: Add an `Debug.Assert(_joinableTaskContext is not null)` in the `JoinableTaskContext.get`?

**Comment by drewnoakes** [comment]
@RussKie &mdash; updates for both your comments pushed.

**Comment by drewnoakes** [comment]
I'll fix the failing tests tomorrow.

**Review by RussKie — COMMENTED** [review]
👍

File: GitExtUtils/GitUI/ThreadHelper.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (drewnoakes) [reply GitExtUtils/GitUI/ThreadHelper.cs:25]: I reverted the changes to this file. I think this is the best approach. We have examples of this in VS. For example, looking up services from MEF. Technically these can return null, but in practice it's super rare and we call this code a lot. So we decided to not annotate it as the cost was too great.

Perhaps the tests could be fixed to not call these when they are null, but I think that is out of scope for this PR. There are many places where the annotations show the truth, and the truth looks weird. We can refactor to a better truth in future.

[the remark this answers is not in the document]
Comment (drewnoakes) [reply GitExtUtils/GitUI/ThreadHelper.cs:25]: Also a debug assert does not instruct the nullability analyzer in the compiler about the nullness of the value.

**Review by RussKie — APPROVED** [review]

File: GitCommands/Git/CommandCache.cs [hunk 3]
```diff
@@ -63,10 +64,10 @@ public IReadOnlyList<string> GetCachedCommands()
         /// <param name="output">Stored output bytes of the command, if found.</param>
         /// <param name="error">Stored error bytes of the command, if found.</param>
         /// <returns><c>true</c> if the command was found, otherwise <c>false</c>.</returns>
-        public bool TryGet(string cmd, [NotNullWhen(returnValue: true)] out byte[]? output, [NotNullWhen(returnValue: true)] out byte[]? error)
+        public bool TryGet(string? cmd, [NotNullWhen(returnValue: true)] out byte[]? output, [NotNullWhen(returnValue: true)] out byte[]? error)
         {
             // Never cache empty commands
-            if (!string.IsNullOrEmpty(cmd))
+            if (!Strings.IsNullOrEmpty(cmd))
```
Comment (gerhardol) [inline GitCommands/Git/CommandCache.cs:70]: Should have realized that this needed to be fixed before pushing...

**Review by gerhardol — COMMENTED** [review]
Dis not see anything in a quick review

File: GitCommands/Git/CommandCache.cs (hunk 3 above)
Comment (drewnoakes) [reply GitCommands/Git/CommandCache.cs:70]: I'm not sure what you mean @gerhardol.

Comment (gerhardol) [reply GitCommands/Git/CommandCache.cs:70]: I did not fix the annotations before pushing.

**Pull request merged**